### PR TITLE
Row select and customized row select with prop feature.

### PR DESCRIPTION
**Pull request by erdembas**
Our purpose of this pull request is to enable row select with pros.

Please review the PR.

We need your feedback to continue forward steps. Documentation, roadmap..

Our roadmap;
we want to make more flexable row select.
* add onRowSelected event
* add onRowDeSelected event
* row click select
* etc.

### CODE REVIEW

File: src/BaseTable.js [hunk 1]
```diff
@@ -49,13 +49,14 @@ class BaseTable extends React.PureComponent {
   constructor(props) {
     super(props);
 
-    const { columns, children, expandedRowKeys, defaultExpandedRowKeys } = props;
+    const { columns, children, expandedRowKeys, defaultExpandedRowKeys, selectedRowKeys } = props;
```
Comment (nihgwu) [inline src/BaseTable.js]: there should be uncontrolled `defaultSelectedRowKeys` as well

File: src/BaseTable.js [hunk 2]
```diff
@@ -1011,6 +1067,14 @@ BaseTable.propTypes = {
...
    */
   defaultExpandedRowKeys: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
+  /**
+   * Selected row keys when initialize the table
+   */
+  selectedRowKeys: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string])),
+  /**
+   * Selected row data key
+   */
+  selectedRowKey: PropTypes.string,
```
Comment (nihgwu) [inline src/BaseTable.js]: this is unnecessary, should be the same as `rowKey`

File: src/BaseTable.js [hunk 3]
```diff
@@ -1011,6 +1067,14 @@ BaseTable.propTypes = {
    * Default expanded row keys when initialize the table
    */
   defaultExpandedRowKeys: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
+  /**
+   * Selected row keys when initialize the table
+   */
+  selectedRowKeys: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string])),
```
Comment (nihgwu) [inline src/BaseTable.js]: the `rowKey` could be `number` now

**Comment by berkayakcay** [comment]
Hi, @nihgwu 
I think that will be good if we support this feature we should guide the people.
Could you share best practise receipt for this feature. 
Would you like to share it on example page?

**Comment by nihgwu** [comment]
@berkayakcay sure, I'll add a demo before we could support that via plugin

**Comment by erdembas** [comment]
@nihgwu We want to clarify that you shared API's are available for your local brach. We couldn't see these API's on code base. Did you share your own API on your code base or you share these image to guide us while completing PR.
What should we do?

**Comment by nihgwu** [comment]
I’ll share this feature as a demo, I shared the api here for early preview or as a guide
Btw that’s not a local branch but a high level component built on BaseTable and used in many products, as I explained before that includes some internal code which is not ready for open source

On May 27, 2019, at 19:10, Erdem BAS <notifications@github.com<mailto:notifications@github.com>> wrote:


@nihgwu<https://github.com/nihgwu> We want to clarify that you shared API's are available for your local brach. We couldn't see these API's on code base. Did you share your own API on your code base or you share these image to guide us while completing PR.
What should we do?

**Comment by berkayakcay** [comment]
We decided to obey your suggestion. We create HOC and use a custom component. We will wait for "plugin support" #3 
Thanks for your kind and fast responses. :1st_place_medal:

**Comment by nihgwu** [comment]
hi @erdembas  @berkayakcay, I've add a [demo](https://autodesk.github.io/react-base-table/docs/selection) for selection feature, sorry for the long wait

**Pull request closed without merging**